okhttp/src/main/java/okhttp3/internal/connection/Transmitter.kt

```diff
@@ -13,309 +13,300 @@
+   * the request completes normally or when it fails due to an exception, in which case [e] should
+   * be non-null.
    *
-   * <p>If the exchange was canceled or timed out, this will wrap {@code e} in an exception that
-   * provides that additional context. Otherwise {@code e} is returned as-is.
+   * If the exchange was canceled or timed out, this will wrap [e] in an exception that provides
+   * that additional context. Otherwise [e] is returned as-is.
    */
-  @Nullable IOException exchangeMessageDone(
-      Exchange exchange, boolean requestDone, boolean responseDone, @Nullable IOException e) {
-    boolean exchangeDone = false;
-    synchronized (connectionPool) {
+  internal fun <E : IOException?> exchangeMessageDone(
+    exchange: Exchange,
+    requestDone: Boolean,
+    responseDone: Boolean,
+    e: E
+  ): E {
+    var result = e
+    var exchangeDone = false
+    synchronized(connectionPool) {
       if (exchange != this.exchange) {
-        return e; // This exchange was detached violently!
+        return result // This exchange was detached violently!
       }
-      boolean changed = false;
+      var changed = false
       if (requestDone) {
-        if (!exchangeRequestDone) changed = true;
-        this.exchangeRequestDone = true;
+        if (!exchangeRequestDone) changed = true
+        this.exchangeRequestDone = true
       }
       if (responseDone) {
-        if (!exchangeResponseDone) changed = true;
-        this.exchangeResponseDone = true;
+        if (!exchangeResponseDone) changed = true
+        this.exchangeResponseDone = true
       }
       if (exchangeRequestDone && exchangeResponseDone && changed) {
-        exchangeDone = true;
-        this.exchange.connection().successCount++;
-        this.exchange = null;
+        exchangeDone = true
+        this.exchange!!.connection()!!.successCount++
+        this.exchange = null
       }
     }
     if (exchangeDone) {
-      e = maybeReleaseConnection(e, false);
+      result = maybeReleaseConnection(result, false)
     }
-    return e;
+    return result
   }
 
-  public @Nullable IOException noMoreExchanges(@Nullable IOException e) {
-    synchronized (connectionPool) {
-      noMoreExchanges = true;
+  fun noMoreExchanges(e: IOException?): IOException? {
+    synchronized(connectionPool) {
+      noMoreExchanges = true
     }
-    return maybeReleaseConnection(e, false);
+    return maybeReleaseConnection(e, false)
   }
 
   /**
    * Release the connection if it is no longer needed. This is called after each exchange completes
    * and after the call signals that no more exchanges are expected.
    *
-   * <p>If the transmitter was canceled or timed out, this will wrap {@code e} in an exception that
-   * provides that additional context. Otherwise {@code e} is returned as-is.
+   * If the transmitter was canceled or timed out, this will wrap [e] in an exception that provides
```

Comment:
``` `e` ``` instead of `[e]`? same next line

Reply:
The benefit of square braces is it participates in refactoring if I rename the parameter.